pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -268,6 +278,28 @@ stdenv.mkDerivation ({
     for f in "$packageConfDir/"*.conf; do
       sed -i "s,dynamic-library-dirs: .*,dynamic-library-dirs: $dynamicLinksDir," $f
     done
+  '') + (optionalString (isCross && setupHaskellDepends != []) ''
```

Comment:
Can you abstract this part out like https://github.com/NixOS/nixpkgs/pull/39735/files#diff-d65852670e35e11b4bd12ed1c634abce ? Fine to wait on separate derivations until #39735 to avoid controversy though :)

Reply:
I thought about it, but I kept the `Setup.hs` changes out of this diff on purpose.